### ISSUE TITLE
Port leaks when retrying connection to SignalR hub.

### ISSUE THREAD

**Issue by roman-76** [SignalR/SignalR#4003]
### Expected  behavior
No growth of bounded ports on client machine when retrying connection to SignalR server
### Actual behavior
amount of bound ports in netstat grows (33872 is PID of the test app) and being exhausted after some time when connection to SignalR retried multiple times 
![netstatoutput](https://user-images.githubusercontent.com/6941230/29889468-1ab730fe-8d79-11e7-8789-160e68c07cc3.png)

After ports exhausted (all free ports bound) no more connections possible from client machine.

### Steps to reproduce
Simple repro program
```
public class AADAuthJobNotificationService
{
    private readonly HubConnection connection;
    public AADAuthJobNotificationService(string notificationEndpoint)
    {
        connection = new HubConnection(notificationEndpoint)
        {
            TraceLevel = TraceLevels.All,
            TraceWriter = Console.Out
        };
    }
    public HubConnection Connection => connection;
    public void Start()
    {
        connection.Start().ContinueWith(t => { if (t.IsFaulted) { Console.WriteLine($"!!!Start Failed {t.Exception}"); } }).Wait();
    }
}

static void Main()
{
    var jns = new AADAuthJobNotificationService("https://nonSignalr.endpoint.net");
    int count = 0;
    do
    {
        count++;
        jns.Start();
        Console.WriteLine($"count={count} State = {jns.Connection.State}");
    }
    while ((Console.ReadLine()) != "Q");
    Console.WriteLine("Completed");
}
```

Program output:
> 16:24:48.8941670 - null - ChangeState(Disconnected, Connecting)
16:24:49.5032479 - null - Disconnected
16:24:49.5042497 - null - Transport.Dispose()
16:24:49.5052495 - null - Closed
!!!Start Failed System.AggregateException: One or more errors occurred. ---> Microsoft.AspNet.SignalR.Client.HttpClientException: StatusCode: 404, ReasonPhrase: 'Not Found', Version: 1.1, Content: System.Net.Http.StreamContent, Headers:
{
  x-content-type-options: nosniff
  Cache-Control: private
  Date: Tue, 18 Jul 2017 16:24:49 GMT
  Server: Microsoft-IIS/10.0
  X-Powered-By: ASP.NET
  Content-Length: 5013
  Content-Type: text/html; charset=utf-8
}
   at Microsoft.AspNet.SignalR.Client.Http.DefaultHttpClient.<>c__DisplayClass5_0.<Get>b__1(HttpResponseMessage responseMessage)
   at Microsoft.AspNet.SignalR.TaskAsyncHelper.<>c__DisplayClass31_0`2.<Then>b__0(Task`1 t)
   at Microsoft.AspNet.SignalR.TaskAsyncHelper.TaskRunners`2.<>c__DisplayClass3_0.<RunTask>b__0(Task`1 t)
   --- End of inner exception stack trace ---
---> (Inner Exception #0) Microsoft.AspNet.SignalR.Client.HttpClientException: StatusCode: 404, ReasonPhrase: 'Not Found', Version: 1.1, Content: System.Net.Http.StreamContent, Headers:
{
  x-content-type-options: nosniff
  Cache-Control: private
  Date: Tue, 18 Jul 2017 16:24:49 GMT
  Server: Microsoft-IIS/10.0
  X-Powered-By: ASP.NET
  Content-Length: 5013
  Content-Type: text/html; charset=utf-8
}
   at Microsoft.AspNet.SignalR.Client.Http.DefaultHttpClient.<>c__DisplayClass5_0.<Get>b__1(HttpResponseMessage responseMessage)
   at Microsoft.AspNet.SignalR.TaskAsyncHelper.<>c__DisplayClass31_0`2.<Then>b__0(Task`1 t)
   at Microsoft.AspNet.SignalR.TaskAsyncHelper.TaskRunners`2.<>c__DisplayClass3_0.<RunTask>b__0(Task`1 t)<---
count=1 State = Disconnected

**Comment by aspnet-hello** [comment]
This issue has been closed as part of issue clean-up as described in https://blogs.msdn.microsoft.com/webdev/2018/09/17/the-future-of-asp-net-signalr/. If you're still encountering this problem, please feel free to re-open and comment to let us know! We're still interested in hearing from you, the backlog just got a little big and we had to do a bulk clean up to get back on top of things. Thanks for your continued feedback!

**Comment by anurse** [comment]
This is the same root cause as https://github.com/SignalR/SignalR/issues/4092 and should be resolved in 2.4.0 when we fix the underlying issue there.
